polar-core/tests/integration_tests.rs

```diff
@@ -1600,21 +1600,29 @@ fn test_and_or_warning() -> TestResult {
 
     p.clear_rules();
     p.load_str("f(x) if x = 1 and x > 1 or x < 3;")?;
-    let msg = p.next_message().unwrap();
-    assert!(matches!(&msg.kind, MessageKind::Warning));
-    assert_eq!(
-        &msg.msg,
-        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages. \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^"
-    );
+    let mut messages = vec![];
+    while let Some(msg) = p.next_message() {
+        messages.push(msg);
+    }
+    assert!(messages.iter().any(|msg| {
+      matches!(&msg.kind, MessageKind::Warning) &&
+      (msg.msg ==
+        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages. \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^")
```

Comment:
nit: Might be mistaken, but I think `assert_eq!()` on a pair of strings (what we used to have here) will give a better error message than this equality check inside an `.any()` inside an `assert!()`

Reply:
Yeah that's definitely true. Can you think of a more idiomatic way to assert that there is "at least one thing in this list meeting this condition"?